xclim/indices/_multivariate.py

```diff
@@ -281,9 +282,10 @@ def daily_temperature_range_variability(
 
        vDTR_j = \frac{ \sum_{i=2}^{I} |(TX_{ij}-TN_{ij})-(TX_{i-1,j}-TN_{i-1,j})| }{I}
     """
+    q = 1 * utils.units2pint(tasmax) - 0 * utils.units2pint(tasmin)
```

Comment:
Previously, the declare_units decorator would impose the output units on the output. I thought it would be less error prone if the decorator rather tried to convert the actual output units to the desired output units instead (see line 379 in utils.py). 

In this case, we have a difference in temperature. If the input is in celcius, the difference will be in celsius. If we convert to Kelvin, we'll subtract 273.15, which does not make sense since it's a difference of temperature, and not an absolute temperature.

This line is used to get the actual units (delta_deg or delta_C) from pint. I've added a check to make sure that the output units are "K" and it works on my end. What version of pint have you checked this with ?

Reply:
> Previously, the declare_units decorator would impose the output units on the output. I thought it would be less error prone if the decorator rather tried to convert the actual output units to the desired output units instead (see line 379 in utils.py).
> 
> In this case, we have a difference in temperature. If the input is in celcius, the difference will be in celsius. If we convert to Kelvin, we'll subtract 273.15, which does not make sense since it's a difference of temperature, and not an absolute temperature.
> 
> This line is used to get the actual units (delta_deg or delta_C) from pint. I've added a check to make sure that the output units are "K" and it works on my end. What version of pint have you checked this with ?

pint == 0.9 when running ```test_static_daily_temperature_range```

Reply:
> pint == 0.9 when running `test_static_daily_temperature_range`

I still get 'kelvin' which version do you have?

Reply:
Sorry this was happening in the xclim.indices call only... when using xclim.atmos module the output units are corrected to 'K'

Reply:
Yes, I see now, fixed anyway.